api/src/main/java/org/openmrs/api/db/hibernate/AuditableInterceptor.java

```diff
@@ -117,8 +118,8 @@ private boolean setCreatorAndDateCreatedIfNull(Object entity, Object[] currentSt
 	private boolean changeProperties(Object[] currentState, String[] propertyNames, boolean objectWasChanged,
 	        Map<String, Object> propertyValues, Boolean setNullOnly) {
 		
-		for (String property : propertyValues.keySet()) {
-			if (changePropertyValue(currentState, propertyNames, property, propertyValues.get(property), setNullOnly)) {
+		for (Map.Entry<String, Object> e : propertyValues.entrySet()) {
```

Comment:
The ticket asked to fix the performance issue caused by use of keyset. I changed that. 
Really optimizing the implementation might be another ticket?